cmake/cpu_extension.cmake

```diff
@@ -68,18 +69,28 @@ if (AVX512_FOUND AND NOT AVX512_DISABLED)
     else()
         message(WARNING "Disable AVX512-BF16 ISA support, no avx512_bf16 found in local CPU flags." " If cross-compilation is required, please set env VLLM_CPU_AVX512BF16=1.")
     endif()
+    
 elseif (AVX2_FOUND)
     list(APPEND CXX_COMPILE_FLAGS "-mavx2")
     message(WARNING "vLLM CPU backend using AVX2 ISA")
+    
 elseif (POWER9_FOUND OR POWER10_FOUND)
     message(STATUS "PowerPC detected")
     # Check for PowerPC VSX support
     list(APPEND CXX_COMPILE_FLAGS
         "-mvsx"
         "-mcpu=native"
         "-mtune=native")
+
+elseif (ASIMD_FOUND)
+    message(STATUS "ARMv8 architecture detected")
+    list(APPEND CXX_COMPILE_FLAGS
+        "-mcpu=native"
+        "-march=armv8.6-a"
```

Comment:
Why was this specific arch chosen?

Reply:
armv8.6-a is the first architecture that supports advance SIMD instructions, bf16 support and SVE support. That's why I chose this specific arch.

Reply:
This is fine for now, but for instance I think Apple's M1 CPU uses ARMv8.4-A so we should consider supporting older versions

Reply:
I see, noted.

Reply:
[According to wikichip](https://en.wikichip.org/wiki/annapurna_labs/graviton/graviton3), Graviton3 is also ARMv8.4A -- is the Graviton3E different? I haven't found any documentation on it

Reply:
@tlrmchlsmth  You're right, graviton 3 has armv8.4. The code needs 8.6 only for bf16 dependencies. But if I just use "-mcpu=native" or "-march=armv8.4-a+bf16" the code works fine on Graviton3 instances. What do you suggest here?

Reply:
If `-march=armv8.4-a+bf16` works, then I suggest using that for this PR.

In general, I think it's best to go with a fairly minimal ISA version, and then explicitly specify what ISA extensions we want to use, especially because ARM has _many_ extensions that are optional for several ISA versions before they become mandatory

Reply:
BTW, the M1 Mac (and I think M2) does not support BF16 NEON so unfortunately this wouldn't help there.

Reply:
I see, thanks for the information. What do you suggest here, to add macos support for future PR or modify into this PR?

Reply:
Hi @mgoin , @tlrmchlsmth I've added the dockerfile and documentation. Changed the arm arch to native. Please have a look.